BrainPortal/lib/s3_connection.rb

```diff
@@ -19,52 +18,244 @@
 # You should have received a copy of the GNU General Public License
 # along with this program.  If not, see <http://www.gnu.org/licenses/>.
 #
-
-require 'aws/s3'
+require 'aws-sdk-s3'
+require 'fileutils'
 
 # A handler connections to Amazon's S3 service.
 class S3Connection
 
   Revision_info=CbrainFileRevision[__FILE__] #:nodoc:
 
-  attr_accessor :base,:service, :bucket, :s3object
+  attr_accessor :resource, :bucket_name, :path_name, :region
```

Comment:
It doesn't look like the class use any of these attributes using methods (e.g. `.bucket_name` or `.bucket_name=` ), so maybe there is no need for attr_accessor? Or on the other hand, the code could be modified such that it doesn't set and get using the instance variables directly.